### PR TITLE
Use `Incomplete` instead of `Any` in `__getattr__`

### PR DESCRIPTION
**Pull request by sobolevn**
I've only touched things that were explicit about being incomplete.
All other `__getattr__` items are untouched.
Refs https://github.com/python/typeshed/pull/8902

### CODE REVIEW

File: stdlib/types.pyi [hunk 1]
```diff
@@ -591,7 +591,7 @@ if sys.version_info >= (3, 9):
             @property
             def __typing_unpacked_tuple_args__(self) -> tuple[Any, ...] | None: ...
 
-        def __getattr__(self, name: str) -> Any: ...  # incomplete
+        def __getattr__(self, name: str) -> Incomplete: ...
```
Comment (JelleZijlstra) [inline stdlib/types.pyi:594]: I think this should actually be Any, as GenericAlias delegates attr lookup to its origin.

Comment (JelleZijlstra) [reply stdlib/types.pyi:594]:
```suggestion
        def __getattr__(self, name: str) -> Any: ...
```

Comment (AlexWaygood) [reply stdlib/types.pyi:594]: Yeah, this one really is inexpressible in the stub

File: stdlib/__main__.pyi [hunk 2]
```diff
@@ -1,3 +1,3 @@
-from typing import Any
+from _typeshed import Incomplete
 
-def __getattr__(name: str) -> Any: ...
+def __getattr__(name: str) -> Incomplete: ...
```
Comment (AlexWaygood) [inline stdlib/__main__.pyi:3]: We want to use the `Incomplete` symbol as a signal for contributors: if a function has `Incomplete` as a parameter or return type, it's us saying "Hey! We'd welcome a PR to fix this thing!"

I'm not sure it's useful to use `Incomplete` here -- is there any way a contributor would be able to improve the stubs for `stdlib/__main__.pyi`?

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
Thanks!

**Pull request merged**